Branch-SDK/src/main/java/io/branch/referral/DeviceInfo.java

```diff
@@ -294,37 +296,54 @@ public String getOsName() {
      * @param context
      * @return user agent string
      */
-    private String getDefaultBrowserAgent(Context context) {
-        String userAgent = "";
+    String getDefaultBrowserAgent(final Context context) {
+        if(!TextUtils.isEmpty(Branch._userAgentString)) {
+            return Branch._userAgentString;
+        }
+
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
             try {
-                // Some devices appear to crash when accessing chromium through the Android framework statics
-                // Suggested alternative is to use a webview instance
-                // https://bugs.chromium.org/p/chromium/issues/detail?id=1279562
-                // https://bugs.chromium.org/p/chromium/issues/detail?id=1271617
-                WebView w = new WebView(context);
-                if(w != null) {
-                    userAgent = w.getSettings().getUserAgentString();
+                PrefHelper.Debug("Did not get user agent string from webview instance. Retrieving from static.");
```

Comment:
@gdeluna-branch 
"Unable to retrieve user agent string from WebView instance. Retrieving from WebSettings"